addons/base_import/models/base_import.py

```diff
@@ -982,6 +1217,121 @@ def do(self, fields, columns, options, dryrun=False):
+
+        Input/output Example:
+            input_file_data: [
+                ["Value part 1", "1", "res.partner_id1", "Value part 2"],
+                ["I am", "1", "res.partner_id1", "Batman"],
+            ]
+            fields: [desc, some_number, partner, desc]
+
+            output merged_data: [
+                ["Value part 1 Value part 2", "1", "res.partner_id1"],
+                ["I am Batman", "1", "res.partner_id1"],
+            ]
+            fields: [desc, some_number, partner]
+        """
+        # Get fields and their occurrences indexes
+        mapping_indexes = collections.defaultdict(list)
+        for idx, field in enumerate(field for field in import_fields if field):
+            mapping_indexes[field].append(idx)
+        import_fields = list(mapping_indexes.keys())
+
+        # recreate data and merge duplicates (applies only on text or char fields)
+        # Also handles multi-mapping on "field of relation fields".
+        merged_data = []
+        for record in input_file_data:
+            new_record = []
+            for fields, indexes in mapping_indexes.items():
+                split_fields = fields.split('/')
+                target_field = split_fields[-1]
+
+                # get target_field type (on target model)
+                target_model = self.res_model
+                for field in split_fields:
+                    if field != target_field:  # if not on the last hierarchy level, retarget the model
+                        target_model = self.env[target_model][field]._name
+                field_type = self.env[target_model].fields_get().get(target_field, {}).get('type', '')
+
+                # merge data if necessary
+                if field_type == 'char':
+                    new_record.append(' '.join(record[idx] for idx in indexes if record[idx]))
+                elif field_type == 'text':
+                    new_record.append('\n'.join(record[idx] for idx in indexes if record[idx]))
+                else:
+                    new_record.append(record[indexes[0]])
+
+            merged_data.append(new_record)
+
+        return import_fields, merged_data
+
+    def _handle_fallback_values(self, import_field, merged_data, fallback_values):
+        """
+        If there are fallback values, this method will replace the input file data value if it does not match the
+        possible values for the given field. This is only valid for boolean and selection fields.
+
+        Note: We can consider that we need to retrieve the selection values for all the fields in fallback_values,
+        as if they are present, it's because there was already a conflict during first import run and user had to
+        select a fallback value for the field.
+
+        :param: (list) import_field: ordered list of field that have been matched to import data
+        :param: (list) merged_data: ordered list of values (list) that need to be imported in the given import_fields
```

Comment:
Would not call it "merged_data" here.
The fact that it has been merged (to handle multi-mapping I presume) is not really relevant for this method.
Maybe simple input_file_data like the previous method?